timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimeLockServiceMetadataIntegrationTest.java

```diff
@@ -0,0 +1,281 @@
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.timelock.api.ConjureIdentifiedVersion;
+import com.palantir.atlasdb.timelock.api.ConjureStartTransactionsRequest;
+import com.palantir.atlasdb.timelock.api.ConjureStartTransactionsResponse;
+import com.palantir.atlasdb.timelock.api.LockWatchRequest;
+import com.palantir.atlasdb.timelock.lock.AsyncLockService;
+import com.palantir.atlasdb.timelock.lock.LockLog;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.lock.AtlasRowLockDescriptor;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.client.IdentifiedLockRequest;
+import com.palantir.lock.client.ImmutableIdentifiedLockRequest;
+import com.palantir.lock.v2.LockResponseV2;
+import com.palantir.lock.v2.LockResponseV2.Successful;
+import com.palantir.lock.v2.LockResponseV2.Unsuccessful;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.watch.ChangeMetadata;
+import com.palantir.lock.watch.LockEvent;
+import com.palantir.lock.watch.LockRequestMetadata;
+import com.palantir.lock.watch.LockWatchCreatedEvent;
+import com.palantir.lock.watch.LockWatchEvent;
+import com.palantir.lock.watch.LockWatchReferences;
+import com.palantir.lock.watch.LockWatchStateUpdate;
+import com.palantir.lock.watch.LockWatchStateUpdate.Snapshot;
+import com.palantir.lock.watch.LockWatchStateUpdate.Success;
+import com.palantir.lock.watch.UnlockEvent;
+import com.palantir.timestamp.InMemoryTimestampService;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import org.awaitility.Awaitility;
+import org.junit.Before;
+import org.junit.Test;
+
+public class AsyncTimeLockServiceMetadataIntegrationTest {
+    private static final TokenVisitor TOKEN_VISITOR = new TokenVisitor();
+    private static final LockWatchEventVisitor LOCK_WATCH_EVENT_VISITOR = new LockWatchEventVisitor();
+    private static final LockWatchStateUpdateVisitor LOCK_WATCH_STATE_UPDATE_VISITOR =
+            new LockWatchStateUpdateVisitor();
+
+    private static final String WATCHED_TABLE_NAME = "watched-table";
+    private static final LockDescriptor WATCHED_LOCK_1 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock1".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_2 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock2".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_3 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock3".getBytes(StandardCharsets.UTF_8));
+    private static final LockDescriptor WATCHED_LOCK_4 =
+            AtlasRowLockDescriptor.of(WATCHED_TABLE_NAME, "lock4".getBytes(StandardCharsets.UTF_8));
```

Comment:
`PtBytes::toBytes`, that way if we change charset encoding we don't need to change it everywhere!